api/src/main/java/org/apache/iceberg/Snapshot.java

```diff
@@ -133,50 +133,76 @@ public interface Snapshot extends Serializable {
+   * Return all data files added to the table in this snapshot.
    * <p>
    * The files returned include the following columns: file_path, file_format, partition,
    * record_count, and file_size_in_bytes. Other columns will be null.
    *
    * @param io a {@link FileIO} instance used for reading files from storage
-   * @return all files added to the table in this snapshot.
+   * @return all data files added to the table in this snapshot.
    */
-  Iterable<DataFile> addedFiles(FileIO io);
+  Iterable<DataFile> addedDataFiles(FileIO io);
 
   /**
-   * Return all files deleted from the table in this snapshot.
+   * Return all data files deleted from the table in this snapshot.
    * <p>
    * The files returned include the following columns: file_path, file_format, partition,
    * record_count, and file_size_in_bytes. Other columns will be null.
    *
-   * @return all files deleted from the table in this snapshot.
-   * @deprecated since 0.14.0, will be removed in 1.0.0; Use {@link Snapshot#deletedFiles(FileIO)} instead.
+   * @return all data files deleted from the table in this snapshot.
+   * @deprecated since 0.14.0, will be removed in 1.0.0; Use {@link Snapshot#deletedDataFiles(FileIO)} instead.
    */
   @Deprecated
   Iterable<DataFile> deletedFiles();
 
   /**
-   * Return all files deleted from the table in this snapshot.
+   * Return all data files deleted from the table in this snapshot.
    * <p>
    * The files returned include the following columns: file_path, file_format, partition,
    * record_count, and file_size_in_bytes. Other columns will be null.
    *
    * @param io a {@link FileIO} instance used for reading files from storage
-   * @return all files deleted from the table in this snapshot.
+   * @return all data files deleted from the table in this snapshot.
    */
-  Iterable<DataFile> deletedFiles(FileIO io);
+  Iterable<DataFile> deletedDataFiles(FileIO io);
+
+  /**
+   * Return all delete files added to the table in this snapshot.
+   * <p>
+   * The files returned include the following columns: file_path, file_format, partition,
+   * record_count, and file_size_in_bytes. Other columns will be null.
+   *
+   * @param io a {@link FileIO} instance used for reading files from storage
+   * @return all delete files added to the table in this snapshot
+   */
+  default Iterable<DeleteFile> addedDeleteFiles(FileIO io) {
+    throw new UnsupportedOperationException(this.getClass().getName() + " doesn't implement addedDeleteFiles");
+  }
+
+  /**
+   * Return all delete files deleted from the table in this snapshot.
+   * <p>
+   * The files returned include the following columns: file_path, file_format, partition,
+   * record_count, and file_size_in_bytes. Other columns will be null.
+   *
+   * @param io a {@link FileIO} instance used for reading files from storage
+   * @return all delete files deleted from the table in this snapshot
+   */
+  default Iterable<DeleteFile> deletedDeleteFiles(FileIO io) {
```

Comment:
Should we use `removed` instead of `deleted` so we don't have `deleted-delete-...`?

Reply:
This question came up a few times. I went with `deleted` for consistency cause we use that for data files in this class and other places like `deletedPositionDeleteFilesCount`. We do use `removed` in snapshot summary but that seems to be the only place.

If we decide to use `removed`, I think we should update the naming for data file methods too. They haven't been released yet. I'd say let's either keep things as they are or switch to this:

```
removedDataFiles(FileIO io);
removedDeleteFiles(FileIO io);
```

Reply:
@rdblue, what do you think?

Reply:
I prefer removed.

Reply:
Sounds good, I'll update both.